pages/data-migration/migrate-from-rdbms-directly.md

```diff
@@ -0,0 +1,281 @@
+
+The `YIELD` keyword will create the `row` objects, which are essentially a map which
+has the column name as the key, and the column value as the value. We can iterate over the row objects with
+the Cypher language to continue populating the database.
+
+```cypher
+CALL migrate.postgresql('Musicians' , {
+  user:'postgres',
+  password:'xxxxxx',
+  host:'localhost',
+  database:'postgresdb'
+})
+YIELD row
+CREATE (n:Musician {id: row.musician_id, name: row.name, instrument: row.instrument});
+```
+
+Similarly, we can inspect the returned rows from the `Bands` table:
+```cypher
+CALL migrate.postgresql('Bands' , {
+  user:'postgres',
+  password:'xxxxxx',
+  host:'localhost',
+  database:'postgresdb'
+})
+YIELD row
+RETURN row.band_id AS band_id, row.band_name AS name;
+```
+
+![](/pages/data-migration/migrate-from-rdbms-directly/migrate-from-rdbms-directly-bands-row.png)
+
+When we're happy with the returned columns, we can create the relationships:
+```cypher
+CALL migrate.postgresql('Bands' , {
+  user:'postgres',
+  password:'xxxxxx',
+  host:'localhost',
+  database:'postgresdb'
+})
+YIELD row
+CREATE (n:Band {id: row.band_id, name: row.band_name});
+```
+
+After we have migrated all the nodes, by issuing a simple query to match all the nodes, we get the following
+disconnected graph.
+```cypher
+MATCH (n) RETURN n;
+```
+
+![](/pages/data-migration/migrate-from-rdbms-directly/migrate-from-rdbms-directly-raw-nodes.png)
+
+### 4. Import relationships from corresponding tables
+We can not do any graph use cases without connections between the nodes, so in this section, we will update the relationships.
+However, in the first argument we will provide a query that selects columns from the table.
+
+```cypher
+CALL migrate.postgresql('SELECT * FROM BandMemberships' , {
+  user:'postgres',
+  password:'xxxxxx',
+  host:'localhost',
+  database:'postgresdb'
+})
+YIELD row
+RETURN row.band_id AS band_id, row.musician_id AS musician_id;
+```
+
+![](/pages/data-migration/migrate-from-rdbms-directly/migrate-from-rdbms-directly-relationships-row.png)
+
+This query is useful as sometimes you don't need all the columns to be migrated to Memgraph, you only need a fraction
+of them. Although you can omit creation of properties when the row is received, this approach consumes less memory on
+the row level. We can create the relationships by matching the source and destination node we migrated previously. Now,
+the indices will play a big role, as the graph can be connected effectively and with optimal performance.
+
+```cypher
+CALL migrate.postgresql('SELECT * FROM BandMemberships' , {
+  user:'postgres',
+  password:'xxxxxx',
+  host:'localhost',
+  database:'postgresdb'
+})
+YIELD row
+MATCH (m:Musician {id: row.musician_id}), (b:Band {id: row.band_id})
+CREATE (m)-[:PERFORMS_IN]->(b);
+```
+
+After migrating the relationships, we can see how our graph looks like:
+
+```cypher
+MATCH (n)-[r]->(m) RETURN n, r, m;
```

Comment:
Avoid using generic variables such as "n" and "m". 
Prefer "m" for musician and "b" for band, as above.